x/group/client/cli/tx.go

```diff
@@ -0,0 +1,650 @@
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/version"
+	authclient "github.com/cosmos/cosmos-sdk/x/auth/client"
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/x/group"
+)
+
+const (
+	FlagExec = "exec"
+	ExecTry  = "try"
+)
+
+// TxCmd returns a root CLI command handler for all x/group transaction commands.
+func TxCmd(name string) *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        name,
+		Short:                      "Group transaction subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	txCmd.AddCommand(
+		MsgCreateGroupCmd(),
+		MsgUpdateGroupAdminCmd(),
+		MsgUpdateGroupMetadataCmd(),
+		MsgUpdateGroupMembersCmd(),
+		MsgCreateGroupAccountCmd(),
+		MsgUpdateGroupAccountAdminCmd(),
+		MsgUpdateGroupAccountDecisionPolicyCmd(),
+		MsgUpdateGroupAccountMetadataCmd(),
+		MsgCreateProposalCmd(),
+		MsgVoteCmd(),
+		MsgExecCmd(),
+	)
+
+	return txCmd
+}
+
+// MsgCreateGroupCmd creates a CLI command for Msg/CreateGroup.
+func MsgCreateGroupCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "create-group [admin] [metadata] [members-json-file]",
+		Short: "Create a group which is an aggregation " +
+			"of member accounts with associated weights and " +
+			"an administrator account. Note, the '--from' flag is " +
+			"ignored as it is implied from [admin].",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Create a group which is an aggregation of member accounts with associated weights and
+an administrator account. Note, the '--from' flag is ignored as it is implied from [admin].
+Members accounts can be given through a members JSON file that contains an array of members.
+
+Example:
+$ %s tx group create-group [admin] [metadata] [members-json-file]
+
+Where members.json contains:
+
+{
+	"members": [
+		{
+			"address": "addr1",
+			"weight": "1",
+			"metadata": "some metadata"
+		},
+		{
+			"address": "addr2",
+			"weight": "1",
+			"metadata": "some metadata"
+		}
+	]
+}
+`,
+				version.AppName,
+			),
+		),
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := cmd.Flags().Set(flags.FlagFrom, args[0])
+			if err != nil {
+				return err
+			}
+
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			members, err := parseMembers(clientCtx, args[2])
+			if err != nil {
+				return err
+			}
+
+			b, err := base64.StdEncoding.DecodeString(args[1])
+			if err != nil {
+				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "metadata is malformed, proper base64 string is required")
+			}
+
+			msg := &group.MsgCreateGroup{
+				Admin:    clientCtx.GetFromAddress().String(),
```

Comment:
This looks like an extra process, can we directly use args[0]

Reply:
we use `cmd.Flags().Set(flags.FlagFrom, args[0])` before